src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -606,9 +608,11 @@ static bool tcp_flush(grpc_exec_ctx* exec_ctx, grpc_tcp* tcp,
 
```

Comment:
Note the above while loop means a partial write went through and you may wan to unref slices in there. Considering it is touching the slice_buffer internals directly, you may not need the new API anyway.

Reply:
This loop won;t leave this function, and so all the unrefs will be picked up later. This is a partial write in that we didn't get all the way through a slice. I am mainly concerned about getting EAGAIN, and having to leave this function then come back and write more later